src/components/sidebar/sidebar.component.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { useTheme } from "@mui/material/styles";
+import Box from "@mui/material/Box";
+import Drawer from "@mui/material/Drawer";
+import CssBaseline from "@mui/material/CssBaseline";
+import Toolbar from "@mui/material/Toolbar";
+import List from "@mui/material/List";
+import Typography from "@mui/material/Typography";
+import Divider from "@mui/material/Divider";
+import IconButton from "@mui/material/IconButton";
+import MenuIcon from "@mui/icons-material/Menu";
+import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import ListItem from "@mui/material/ListItem";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText from "@mui/material/ListItemText";
+import { Dashboard, AttachMoney, Receipt, History } from "@mui/icons-material";
+import { useNavigate } from "react-router";
+import { createSideBarLink } from "./helper";
+import { Main, AppBar, DrawerHeader } from "./sidebar.styles";
+import { drawerWidth } from "./constant";
+
+const sidebarLinks = [
+  createSideBarLink("Dashboard", "/dashboard", <Dashboard />),
+  createSideBarLink("Funds Transfer", "/transfer", <AttachMoney />),
+  createSideBarLink("Transaction History", "/history", <History />),
+  createSideBarLink("Utility Bills", "/bills", <Receipt />),
+];
+
+const Sidebar = ({ children }: { children: any }) => {
+  const theme = useTheme();
+  const [open, setOpen] = useState(false);
+  const navigate = useNavigate();
+
+  const handleDrawerOpen = () => setOpen(true);
+
+  const handleDrawerClose = () => setOpen(false);
+
+  return (
+    <Box sx={{ display: "flex" }}>
+      <CssBaseline />
+      <AppBar position="fixed" open={open}>
+        <Toolbar>
+          <IconButton
+            color="inherit"
+            aria-label="open drawer"
+            onClick={handleDrawerOpen}
+            edge="start"
+            sx={{ mr: 2, ...(open && { display: "none" }) }}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography variant="h6" noWrap component="div">
+            My Bank App
+          </Typography>
+        </Toolbar>
+      </AppBar>
+      <Drawer
+        sx={{
+          width: drawerWidth,
+          flexShrink: 0,
+          "& .MuiDrawer-paper": {
+            width: drawerWidth,
+            boxSizing: "border-box",
+          },
+        }}
+        variant="persistent"
+        anchor="left"
+        open={open}
+      >
+        <DrawerHeader>
+          <IconButton onClick={handleDrawerClose}>
+            {theme.direction === "ltr" ? (
+              <ChevronLeftIcon />
+            ) : (
+              <ChevronRightIcon />
+            )}
+          </IconButton>
+        </DrawerHeader>
+        <Divider />
+        <List>
+          {sidebarLinks.map(({ name, path, icon }, index) => (
```

Comment:
name, path and icon are being passed as function parameters, why are you de-structuring them?